pandas/core/arrays/arrow/array.py

```diff
@@ -1729,42 +1730,18 @@ def _replace_with_mask(
 
     def _to_masked(self):
         pa_dtype = self._pa_array.type
-        na_value = 1
-        from pandas.core.arrays import (
-            BooleanArray,
-            FloatingArray,
-            IntegerArray,
-        )
-
-        arr_cls: type[FloatingArray | IntegerArray | BooleanArray]
-        if pa.types.is_floating(pa_dtype):
-            nbits = pa_dtype.bit_width
-            dtype = f"Float{nbits}"
-            np_dtype = dtype.lower()
-            arr_cls = FloatingArray
-        elif pa.types.is_unsigned_integer(pa_dtype):
-            nbits = pa_dtype.bit_width
-            dtype = f"UInt{nbits}"
-            np_dtype = dtype.lower()
-            arr_cls = IntegerArray
-
-        elif pa.types.is_signed_integer(pa_dtype):
-            nbits = pa_dtype.bit_width
-            dtype = f"Int{nbits}"
-            np_dtype = dtype.lower()
-            arr_cls = IntegerArray
 
+        if pa.types.is_floating(pa_dtype) or pa.types.is_integer(pa_dtype):
+            na_value = 1
         elif pa.types.is_boolean(pa_dtype):
-            dtype = "boolean"
-            np_dtype = "bool"
             na_value = True
-            arr_cls = BooleanArray
         else:
             raise NotImplementedError
 
+        dtype = _arrow_dtype_mapping()[pa_dtype]
```

Comment:
could we simplify even more by just doing `self.dtype.numpy_dtype`?  Aside from that, LGTM

Reply:
I'd prefer keeping the mask stuff, this should make the conversion faster

Reply:
Sounds good.